### DISCUSSION TITLE
`new PrismaClient()` returning `undefined` in Next.js

### DISCUSSION THREAD

**Discussion by joelwright-dev** [prisma/prisma#23172] [Q&A]
### Bug description

As stated in the title, I am using `prisma = new PrismaClient()` according to the documentation. I have modified this using a pretty standard template for creating a global prisma instance with Next.js as shown below:

```// lib/prisma.ts
import { PrismaClient } from "@prisma/client";

let prisma: PrismaClient;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

When I try to use this prisma instance using `import prisma from "../lib/prisma";` and then using getStaticProps:

```
export async function getStaticProps() {
  const categories = await prisma.category.findMany();
  const items = await prisma.item.findMany();

  console.log(categories);
  console.log(items);
  console.log(prisma);

  return {
    props: { categories, items, prisma },
  };
}
```

When this is logged to console all variables return as undefined.

Any insight as to why this is returning undefined would be much appreciated. Thank you.

Versions
next - 14.0.4
prisma - 5.7.1
@prisma/client - 5.7.1

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model Item {
  id          Int       @id @default(autoincrement())
  name        String
  description String
  Category    Category? @relation(fields: [categoryId], references: [id])
  categoryId  Int?
}

model Category {
  id          Int    @id @default(autoincrement())
  name        String
  description String
  icon        String
  items       Item[]
}
```

```ts
// lib/prisma.ts
import { PrismaClient } from "@prisma/client";

let prisma: PrismaClient;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;

```

### Environment & setup

- OS: Linux 6.6.7-arch1-1 x86_64
- Database: MySQL
- Node.js version: v21.5.0

### Prisma Version

```
prisma                  : 5.7.1
@prisma/client          : 5.7.1
Computed binaryTarget   : debian-openssl-3.0.x
Operating System        : linux
Architecture            : x64
Node.js                 : v21.5.0
Query Engine (Node-API) : libquery-engine 0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5 (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.7.1-1.0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5
Default Engines Hash    : 0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5
Studio                  : 0.495.0
```

**Comment by ludralph** [comment]
Hi @joelwright-dev 👋 

Thank you for raising this question.

Did you run `npx prisma generate` after making your changes to your schema file? Also you may also want to update your `lib/prisma.ts`  as recommended in this section of the [documentation](https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/nextjs-prisma-client-dev-practices#solution).
```ts
import { PrismaClient } from '@prisma/client'

const prismaClientSingleton = () => {
  return new PrismaClient()
}

declare global {
  var prisma: undefined | ReturnType<typeof prismaClientSingleton>
}

const prisma = globalThis.prisma ?? prismaClientSingleton()

export default prisma

if (process.env.NODE_ENV !== 'production') globalThis.prisma = prisma
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
